File: MATLAB and R_Programs/diff_ver.R
# function to calculate the first order difference matrix vertically
# row is the number of the pixel along the vertical direction
# col is the number of pixel along the horizontal durection
# the total pixel will be row*col
diff_ver=function(row,col){
  c=rep(0,row*col);c[1]=1;c[col+1]=-1
  L=toeplitz(c);L[upper.tri(L)]=0
  return(list(L))}

File: MATLAB and R_Programs/diff_ver.py
import numpy as np

def diff_ver(row, col):
    # Function to compute 1st order difference along the vertical
    c = np.zeros(row * col)
    c[0] = 1
    c[col] = -1
    L2 = np.zeros((row*col, row*col))
    for i in range(row*col):
        L2[i, i:] = c[:row*col - i]
    return L2
